patches/net/minecraft/core/NonNullList.java.patch

```diff
@@ -0,0 +1,25 @@
+--- a/net/minecraft/core/NonNullList.java
++++ b/net/minecraft/core/NonNullList.java
+@@ -9,6 +_,22 @@
+ import org.apache.commons.lang3.Validate;
+ 
+ public class NonNullList<E> extends AbstractList<E> {
++
++    /**
++     * Neo: utility method to construct a Codec for a NonNullList
++     * @param entryCodec The codec to use for the elements
++     * @param <E> The element type
++     * @return A codec that encodes as a list, and decodes into NonNullList
```

Comment:
```suggestion
+     * @param entryCodec the codec to use for the elements
+     * @param <E> the element type
+     * @return a codec that encodes as a list, and decodes into NonNullList
```

Reply:
Ah right. That always throws me off. Someday I have to figure out why those descriptions shouldn't be capitalized.